### ISSUE TITLE
开启了reuse_port报nginx: [emerg] duplicate listen options for 0.0.0.0:443

### ISSUE THREAD

**Issue by javais** [alibaba/tengine#868]
nginx: [emerg] duplicate listen options for 0.0.0.0:443 in /usr/local/nginx/conf/nginx.conf:155

**Comment by Shuliyey** [comment]
**/usr/local/nginx/conf/nginx.conf**

因该之前已经有个server 在接听443 ssl 端口。

可以上传下 **/usr/local/nginx/conf/nginx.conf** 配置吗

**Comment by fullstack-dev** [comment]
user  nobody;

worker_processes auto;
worker_cpu_affinity auto;
worker_rlimit_nofile 65535;

error_log  logs/error.log  error;

pid        logs/nginx.pid;

events {
	use epoll;
	multi_accept on;
	reuse_port on;
	worker_connections  65535;
	accept_mutex off;
}


http {
    include       mime.types;
    default_type  application/octet-stream;

    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$http_x_forwarded_for"';

	error_log  logs/error.log crit;
    access_log  logs/access.log  main;

    sendfile        on;
    tcp_nopush     on;
    keepalive_timeout  10;
	client_body_timeout 10;
	reset_timedout_connection on;
	send_timeout 60;
	
	server_names_hash_bucket_size 128;
	large_client_header_buffers 4 4k;
	client_max_body_size 100m;
	client_header_buffer_size 4k;
	client_body_buffer_size 256k;
	
    gzip on;
	gzip_disable "msie6";
	gzip_proxied any;
	gzip_min_length 1000;
	gzip_comp_level 6;
	gzip_types text/plain text/css application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
	open_file_cache max=100000 inactive=20s;
	open_file_cache_valid 30s;
	open_file_cache_min_uses 2;
	open_file_cache_errors on;
	
	proxy_cache_path /usr/local/nginx/proxy_cache levels=1:2 keys_zone=cache:20m inactive=1d max_size=100m;
	
	map $http_upgrade $connection_upgrade {
		default upgrade;
		'' close;
	}
	
	dyups_upstream_conf  upstream.conf;
    include upstream.conf;
	
	req_status_zone server "$host,$server_addr:$server_port" 10M;
    req_status_zone_add_indicator server $limit;
	
	server {
		listen 80;
		server_name *.example.me;
		add_header Strict-Transport-Security "max-age=31536000;includeSubDomains";
		rewrite ^(.*)$ https://$host$1 permanent;
	}
	
	server {
		listen 443 ssl ssl http2 fastopen=3;
		server_name  www.example.me;
		
		add_header Strict-Transport-Security "max-age=31536000;includeSubDomains";
		ssl on;
        ssl_certificate ../ssl/fullchain.pem;
        ssl_certificate_key ../ssl/privkey.pem;
		
		ssl_dhparam ../ssl/dhparam.pem;
		
		ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
		ssl_ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-RC4-SHA:!ECDHE-RSA-RC4-SHA:ECDH-ECDSA-RC4-SHA:ECDH-RSA-RC4-SHA:ECDHE-RSA-AES256-SHA:!RC4-SHA:HIGH:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!CBC:!EDH:!kEDH:!PSK:!SRP:!kECDH;
		
		ssl_prefer_server_ciphers on;
		ssl_session_timeout 12m;
		ssl_session_cache shared:SSL:16m;
		ssl_buffer_size 8k;
		ssl_session_tickets on;
		ssl_stapling on;
		ssl_stapling_verify on;
		resolver 8.8.4.4 8.8.8.8 valid=300s;
		resolver_timeout 10s;
		
		
		proxy_redirect off;
		
		client_body_buffer_size 256k;
		proxy_connect_timeout 5;
		proxy_send_timeout 300;
		proxy_read_timeout 300;
		proxy_buffer_size 64k;
		proxy_buffers 32 32k;
		proxy_busy_buffers_size 128k;
		proxy_temp_file_write_size 64k;
		
		set $web_host web;
		
		location / {
			proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
			proxy_set_header X-Real-IP $remote_addr;
			proxy_set_header Host $http_host;
			proxy_redirect off;
			proxy_pass http://$web_host;
		}
		
		set $limit 0;
        if ($arg_limit = '1') {
            set $limit 1;
        }
        req_status server;
	}
	
	server {
		listen 443 ssl;
		server_name  passport.example.me;
		
		add_header Strict-Transport-Security "max-age=31536000;includeSubDomains";
		ssl on;
        ssl_certificate ../ssl/fullchain.pem;
        ssl_certificate_key ../ssl/privkey.pem;
		
		ssl_dhparam ../ssl/dhparam.pem;
		
		ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
		ssl_ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-RC4-SHA:!ECDHE-RSA-RC4-SHA:ECDH-ECDSA-RC4-SHA:ECDH-RSA-RC4-SHA:ECDHE-RSA-AES256-SHA:!RC4-SHA:HIGH:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!CBC:!EDH:!kEDH:!PSK:!SRP:!kECDH;
		
		ssl_prefer_server_ciphers on;
		ssl_session_timeout 12m;
		ssl_session_cache shared:SSL:16m;
		ssl_buffer_size 8k;
		ssl_session_tickets on;
		ssl_stapling on;
		ssl_stapling_verify on;
		resolver 8.8.4.4 8.8.8.8 valid=300s;
		resolver_timeout 10s;
		
		
		proxy_redirect off;
		client_max_body_size 100M;
		client_body_buffer_size 256k;
		proxy_connect_timeout 5;
		proxy_send_timeout 300;
		proxy_read_timeout 300;
		proxy_buffer_size 64k;
		proxy_buffers 32 32k;
		proxy_busy_buffers_size 128k;
		proxy_temp_file_write_size 64k;
		
		set $passport_host passport;
		
		location / {
			proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
			proxy_set_header X-Real-IP $remote_addr;
			proxy_set_header Host $http_host;
			proxy_redirect off;
			proxy_pass http://$passport_host;
		}
		
		set $limit 0;
        if ($arg_limit = '1') {
            set $limit 1;
        }
        req_status server;
	}
	
	server {
		listen 443 ssl;
		server_name  img.example.me;
		
		add_header Strict-Transport-Security "max-age=31536000;includeSubDomains";
		ssl on;
        ssl_certificate ../ssl/img.example.me/fullchain.pem;
        ssl_certificate_key ../ssl/img.example.me/privkey.pem;
		
		ssl_dhparam ../ssl/dhparam.pem;
		
		
		ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
		ssl_ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-RC4-SHA:!ECDHE-RSA-RC4-SHA:ECDH-ECDSA-RC4-SHA:ECDH-RSA-RC4-SHA:ECDHE-RSA-AES256-SHA:!RC4-SHA:HIGH:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!CBC:!EDH:!kEDH:!PSK:!SRP:!kECDH;
		
		ssl_prefer_server_ciphers on;
		ssl_session_timeout 12m;
		ssl_session_cache shared:SSL:16m;
		ssl_buffer_size 8k;
		ssl_session_tickets on;
		ssl_stapling on;
		ssl_stapling_verify on;
		resolver 8.8.4.4 8.8.8.8 valid=300s;
		resolver_timeout 10s;
		
		
		proxy_redirect off;
		proxy_connect_timeout 5;
		proxy_send_timeout 300;
		proxy_read_timeout 300;
		proxy_buffer_size 64k;
		proxy_buffers 32 32k;
		proxy_busy_buffers_size 128k;
		proxy_temp_file_write_size 64k;
		
		set $ambry_host ambry;
		location / {
			proxy_method GET;
			proxy_pass http://$ambry_host;
		}
		
		set $limit 0;
        if ($arg_limit = '1') {
            set $limit 1;
        }
        req_status server;
	}
	
	server {
		listen 443 ssl;
		server_name  cdn.example.me;
		
		add_header Strict-Transport-Security "max-age=31536000;includeSubDomains";
		ssl on;
        ssl_certificate ../ssl/fullchain.pem;
        ssl_certificate_key ../ssl/privkey.pem;
		
		ssl_dhparam ../ssl/dhparam.pem;
		
		
		ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
		ssl_ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-RC4-SHA:!ECDHE-RSA-RC4-SHA:ECDH-ECDSA-RC4-SHA:ECDH-RSA-RC4-SHA:ECDHE-RSA-AES256-SHA:!RC4-SHA:HIGH:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!CBC:!EDH:!kEDH:!PSK:!SRP:!kECDH;
		
		ssl_prefer_server_ciphers on;
		ssl_session_timeout 12m;
		ssl_session_cache shared:SSL:16m;
		ssl_buffer_size 8k;
		ssl_session_tickets on;
		ssl_stapling on;
		ssl_stapling_verify on;
		resolver 8.8.4.4 8.8.8.8 valid=300s;
		resolver_timeout 10s;
		
		
		proxy_redirect off;
		proxy_connect_timeout 5;
		proxy_send_timeout 300;
		proxy_read_timeout 300;
		proxy_buffer_size 64k;
		proxy_buffers 32 32k;
		proxy_busy_buffers_size 128k;
		proxy_temp_file_write_size 64k;
		
		set $passport_host passport;
		
		location ~ .*\.(html|htm|gif|jpg|jpeg|bmp|png|ico|txt|js|css)$ {
			root html;
			index index.html;
			expires      7d;
			access_log off;
			proxy_store on;
			proxy_temp_path html;
			proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
			proxy_set_header X-Real-IP $remote_addr;
			proxy_set_header Host $http_host;
			proxy_set_header X-Real-IP $remote_addr;
			if ( !-e $request_filename ) {
				proxy_pass http://$passport_host;
			}
		}
		
		set $limit 0;
        if ($arg_limit = '1') {
            set $limit 1;
        }
        req_status server;
	}
}

**Comment by Shuliyey** [comment]
第一个 

```
server {
	listen 443 ssl http2 fastopen=3;
	server_name  www.example.me;
```

可以保持不变 或者保险可以改成


后面所有的 把 ssl 部分删了, 只要 listen 443
```
server {
	listen 443

**Comment by chobits** [comment]
Has been fixed in v2.3.0.

Tengine-2.3.0 has been released, it has been been upgraded to nginx 1.15.9.

For more details of new version, see https://github.com/alibaba/tengine/issues/1220.
